src/main/java/com/tsunderebug/speedrun4j/exception/SpeedrunUrlException.java

```diff
@@ -0,0 +1,9 @@
+package com.tsunderebug.speedrun4j.exception;
+
+public class SpeedrunUrlException extends RuntimeException{
+
+    public SpeedrunUrlException(String message) {
```

Comment:
This should include the URL that failed, and augment the message.

Malformed URLs and failed connections should be two separate `Exceptions`.

The former should be an `IllegalArgumentException` and the latter should be an `IOException`.

Reply:
Alright thanks for letting me know